oauthproxy.go

```diff
@@ -365,21 +366,40 @@ func (p *OAuthProxy) MakeCSRFCookie(req *http.Request, value string, expiration
 }
 
 func (p *OAuthProxy) makeCookie(req *http.Request, name string, value string, expiration time.Duration, now time.Time) *http.Cookie {
-	if p.CookieDomain != "" {
-		domain := req.Host
+	cookieDomain := ""
+	host := req.Header.Get("X-Forwarded-Host")
+	if host == "" {
+		host = req.Host
+	}
+
+	// Sort cookie domains by length, so that we try longer (and more specific)
+	// domains first
+	sortedDomains := p.CookieDomains
```

Comment:
Perhaps the sorting could be done ahead of time rather than for each time we request to make a cookie? It could be a private field on the `OAuthProxy` that then get's populated as the proxy start up, maybe during validation? WDYT?